Translate Glossary to Indonesian (#36)
reactjs/id.reactjs.org: content/docs/reference-glossary.md

---

## Aplikasi Single-page {#single-page-application}

> Aplikasi _Single-page_ adalah jenis aplikasi web yang memuat sebuah halaman HTML dan seluruh aset yang diperlukannya (seperti Javascript dan CSS) untuk dapat berjalan. Seluru interaksi yang terjadi pada halaman yang telah dimuat maupun halaman lainnya yang berhubungan, tidak memerlukan tambahan pengambilan data ke server, yang berarti halaman tidak perlu dimuat ulang.

Reviewer: Seluru -> Seluruh I think it's a typo 😄

Aplikasi _Single-page_ tidak harus dibangun menggunakan React. React dapat digunakan untuk meningkatkan bagian-bagian tertentu dari halaman yang telah ada untuk meningkatkan interaktivitas halaman tersebut. Kode yang ditulis dalam React dapat berjalan dengan aman dengan jenis lainnnya yang di _render_ pada _server_, seperti PHP, atau _library_ lain yang ada di sisi _client_. Faktanya, dengan cara inilah Facebook menggunakan React.

> ## ES6, ES2015, ES2016, etc {#es6-es2015-es2016-etc}

Reviewer: etc -> dsb

What do you think? 😄

> Semua akronim diatas merujuk pada versi terbaru dari standar Spesifikasi bahasa ECMAScript, yang merupakan implementasi dari bahasa JavaScript Versi ES6 (dikenal juga sebagai ES2015), yang mencakup tambahan dari versi sebelumnya seperti : _Arrow Function_, kelas, _template literals_, deklarasi variabel dengan let dan const. Anda dapat mempelajarinya lebih dalam tiap versinya melalui [tautan ini](https://en.wikipedia.org/wiki/ECMAScript#Versions)

Reviewer: `let` dan `const` looks like missing backticks 😄

## Kompiler {#compilers}

> Sebuah kompiler JavaScript mengambil, dan mengubah kode JavaScript, dan mengembalikannya kedalam format yang berbeda. Kasus paling umum penggunaan compiler adalah saat sintaksis ES6 diubah untuk dapat dimengerti oleh peramban tipe lama. [Babel](https://babeljs.io/) adalah salah satu jenis kompiler yang sering digunakan dengan React.

Reviewer: ```suggestion
Sebuah kompiler JavaScript menerima kode JavaScript, mengubahnya, dan mengembalikannya kedalam format yang berbeda. Kasus paling umum penggunaan kompiler adalah untuk mengubah sintaksis ES6 ke sintaksis yang dapat dimengerti oleh peramban versi lama. [Babel](https://babeljs.io/) adalah salah satu jenis kompiler yang sering digunakan dengan React.
```
What do you think? 😄

## Bundler {#bundlers}

> Bundler menggunakan kode JavaScript dan CSS yang ditulis di modul terpisah, dan menggabungkannya menjadi beberapa file yang lebih optimal untuk dijalankan oleh peramban. Bundler yang sering digunakan React meliputi [Webpack](https://webpack.js.org/) dan [Browserify](http://browserify.org/)

Reviewer: Bundler is a loanword so needs to be italicized (in the body of paragraph) *Bundler* 😄

> _Bundler_ menggunakan kode JavaScript dan CSS yang ditulis di modul terpisah, dan menggabungkannya menjadi beberapa file yang lebih optimal untuk dijalankan oleh peramban. _Bundler_ yang sering digunakan React meliputi [Webpack](https://webpack.js.org/) dan [Browserify](http://browserify.org/)

Reviewer 2: Browser need to translate to peramban?

Reviewer 3: Other docs doesn't change `browser` to `peramban`. Here are some references from the `master` branch:
- [faq-functions.md#L310](https://github.com/reactjs/id.reactjs.org/blob/master/content/docs/faq-functions.md#L310)
- [handling-events.md#L57](https://github.com/reactjs/id.reactjs.org/blob/master/content/docs/handling-events.md#L57)
- [reference-javascript-environment-requirements.md#L9](https://github.com/reactjs/id.reactjs.org/blob/master/content/docs/reference-javascript-environment-requirements.md#L9)

> ## Manajer Paket {#package-managers}

Reviewer: Hmm this one is though, Manajer Paket is a valid translation for Package Managers, but I don't think it sounds familiar to the readers, what's your opinion on this guys @gedeagas @arfianadam ?

Reviewer 2: Don't translate it, I guess @regalius . Just use Package Managers

Reviewer 3: Stick to "Package Manager" 👍

> Manajer paket adalah alat yang digunakan untuk mengatur dan mengelola dependensi pada project Anda. [npm](https://www.npmjs.com/) dan [Yarn](http://yarnpkg.com/) adalah manajer paket yang paling sering digunakan pada aplikasi React dimana keduanya adalah klien yang memiliki registri paket npm yang sama.

Reviewer: project -> proyek

## CDN {#cdn}

> CDN adalah singkatan dari Content Delivery Network. CDN menyediakan konten yang statis dan di-_cache_ dari jaringan server diseluruh dunia.

Reviewer: Content Delivery Network -> *Content Delivery Network*

## JSX {#jsx}

> JSX adalah sintaksis ekstensi untuk JavaScript. JSX mirip dengan bahasa _template_, namum memiliki kekuatan penuh atas bahasa JavaScript. JSX dijalankan menjadi format `React.createElement()` yang mengembalikan objek JavaScript baru berupa "Elemen React". Untuk mendapatkan penjelasan dasar mengenai mengenai JSX dapat melihat [dokumentasi JSX](/docs/introducing-jsx.html), atau mengenai JSX dengan lebih mendalam [di sini](https://reactjs.org/docs/introducing-jsx.html)

Reviewer: namum -> namun I think typo 😄 

```
JSX gets compiled to `React.createElement()` calls which return plain JavaScript objects called "React elements"
```
```
JSX akan dikompilasi ke pemanggilan `React.createElement()` yang mengembalikan objek JavaScript biasa bernama "React elements"
```

What do you think? 😄

> Tidak seperti dalam HTML, React DOM menggunakan format camelCase untuk penamaan atributnya. Sebagai contoh, `tabindex` menjadi `tabIndex` di JSX. Atribut `class` juga ditulis sebagai `className` karena `class` adalah nama yang telah dipesan dalam JavaScript.

Reviewer: I think the context of this sentence

```
React DOM uses camelCase property naming convention instead of HTML attribute names. 
```
Is closer to this
```
Alih-alih menggunakan nama atribut HTML, React DOM menggunakan konvensi penamaan *camelCase* untuk propertinya
```

What do you think? 😄

```js
const name = 'Clementine';
ReactDOM.render(
  <h1 className="hello">Nama saya {name}!</h1>,
  document.getElementById('root')
);
```  

## [Elemen](/docs/rendering-elements.html) {#elements}

> React Elemen adalah bagian pembangun aplikasi React. Beberapa orang bisa jadi salah mengartikan elemen sebagai istilah 'komponen' yang lebih dikenal luas. Sebuah elemen menjelaskan apa yang ingin Anda tampilkan di layar. Elemen React tidak dapat diubah (immutable).

Reviewer: ```suggestion
Elemen React adalah bagian pembangun aplikasi React. Beberapa orang bisa jadi salah mengartikan elemen sebagai istilah 'komponen' yang lebih dikenal luas. Sebuah elemen menjelaskan apa yang ingin Anda tampilkan di layar. Elemen React tidak dapat diubah (*immutable*).
```

}
```

> Komponen dapat dipecah menjadi beberapa bagian fungsi berbeda yang dapat digunakan dengan komponen lainnya. Komponen dapat mengembalikan array, string, angka, maupun mengembalikan komponen lainnya. Aturan dasar yang praktis yaitu, jika ada bagian dari antarmuka Anda yang digunakan beberapa kali (Tombol, Panel, Avatar), atau cukup kompleks untuk berdiri dengan sendirinya (App, FeedStory, Komentar), akan lebih baik jika bagian tersebut menggunakan Reusable Component. Nama komponen juga harus selalu diawali dengan huruf kapital (`<Wrapper/>` **bukan** `<wrapper/>`). Lihat [dokumentasi](https://reactjs.org/docs/components-and-props.html#rendering-a-component) untuk penjelasan mengenai cara melakukan _render_ komponen.

Reviewer: array -> senarai
string -> *string*

```suggestion
Komponen dapat dipecah menjadi beberapa bagian fungsi berbeda yang dapat digunakan dengan komponen lainnya. Komponen dapat mengembalikan senarai, *string*, angka, maupun mengembalikan komponen lainnya. Aturan dasar yang praktis yaitu, jika ada bagian dari antarmuka pengguna Anda yang digunakan beberapa kali (Tombol, Panel, Avatar), atau cukup kompleks untuk berdiri dengan sendirinya (App, FeedStory, Komentar), komponen-komponen tersebut adalah kandidat yang cocok untuk dijadikan komponen yang dapat digunakan kembali (*reusable component*). Nama komponen juga harus selalu diawali dengan huruf kapital (`<Wrapper/>` **bukan** `<wrapper/>`). Lihat [dokumentasi ini](https://reactjs.org/docs/components-and-props.html#rendering-a-component) untuk penjelasan mengenai cara melakukan _render_ komponen.
```
What do you think? 😄

### [`props`](/docs/components-and-props.html) {#props}

`props` adalah masukan dari komponen React. `props` adalah data yang dioper dari _parent component_ ke _child component_-nya.

> Perlu diingat bahwa `props` adalah _readonly_, sehingga nilainya tidak dapat diubah melalui cara apapun:

Reviewer: ```suggestion
Perlu diingat bahwa `props` bersifat _readonly_, sehingga nilainya tidak dapat diubah melalui cara apapun:
```

Untuk setiap bagian data yang berubah, harus terdapat satu komponen yang 'memiliki' data tersebut sebagai _state_ di dalamnya. Jangan mencoba untuk melakukan sinkronisasi _state_ pada dua komponen yang berbeda. Sebaliknya, [angkat nilainya ke atas](/docs/lifting-state-up.html), kemudian oper sebagai _props_ untuk kedua komponen tersebut.

## [Lifecycle Methods](/docs/state-and-lifecycle.html#adding-lifecycle-methods-to-a-class) {#lifecycle-methods}

_Lifecycle Methods_ adalah fungsionalitas khusus yang dijalankan selama fase tertentu dalam sebuah komponen. Terdapat beberapa metode yang tersedia saat komponen dibuat dan dimasukkan kedalam DOM ([mounting](/docs/react-component.html#mounting)), saat komponen diperbarui, dan saat komponen dilepas atau dihapus dari DOM.

 ## [Controlled](/docs/forms.html#controlled-components) vs. [Uncontrolled Components](/docs/uncontrolled-components.html)

> React memiliki dua pendekatan berbeda untuk mengatur masukan formulir.

Reviewer 2: Is that necessary to translate form into formulir?

Reviewer 2: what do you think @regalius?

Reviewer 3: ```suggestion
React memiliki dua pendekatan berbeda untuk mengatur inputan form.
```

Maybe this?

Reviewer 3: `inputan form` sounds better IMO.

### [`state`](/docs/state-and-lifecycle.html#adding-local-state-to-a-class) {#state}

> Sebuah komponen membutuhkan `state` saat sebuah data yang berhubungan dengannya berubah sewaktu-waktu. Sebagia contoh, sebuah komponen `Checkbox` dapat membutuhkan _state_ `isChecked`, dan sebuah komponen `NewsFeed` bisa jadi selalu membutuhkan _state_ `fetchedPost` terbaru.

Reviewer: Sebagia -> Sebagian typo 😄

Perbedaan paling penting diantara `state` dan `props` adalah `props` dioper melalui _parent component_, sedangkan `state` diatur oleh komponen itu sendiri. Sebuah komponen tidak dapat mengubah `props`, tetapi dapat mengubah `state`. Untuk melakukannya, komponen tersebut harus memanggil fungsi `this.setState()`. Hanya komponen yang didefinisikan sebagai kelas yang dapat memiliki _state_.

> Untuk setiap bagian data yang berubah, harus terdapat satu komponen yang 'memiliki' data tersebut sebagai _state_ di dalamnya. Jangan mencoba untuk melakukan sinkronisasi _state_ pada dua komponen yang berbeda. Sebaliknya, [lift it up](/docs/lifting-state-up.html), atau angkat nilainya ke arah pendahulunya yang terdekat, kemudian oper sebagai _props_ untuk kedua komponen tersebut.

Reviewer: ```suggestion
Untuk setiap bagian data yang berubah, harus terdapat satu komponen yang 'memiliki' data tersebut sebagai _state_ di dalamnya. Jangan mencoba untuk melakukan sinkronisasi _state_ pada dua komponen yang berbeda. Sebaliknya, [angkat nilainya ke atas](/docs/lifting-state-up.html), kemudian oper sebagai _props_ untuk kedua komponen tersebut.
```

> ## [Metode Lifecycle](/docs/state-and-lifecycle.html#adding-lifecycle-methods-to-a-class) {#lifecycle-methods}

Reviewer: Lifecycle Methods doesn't have to be translated since it is a React term 😄

Metode lifecycle adalah fungsionalitas khusus yang dijalankan selama fase tertentu dalam sebuah komponen. Terdapat beberapa metode yang tersedia saat komponen dibuat dan dimasukkan kedalam DOM ([mounting](/docs/react-component.html#mounting)), saat komponen diperbarui, dan saat komponen dilepas atau dihapus dari DOM.

 ## [Controlled](/docs/forms.html#controlled-components) vs. [Uncontrolled Components](/docs/uncontrolled-components.html)

React memiliki dua pendekatan berbeda untuk mengatur masukan formulir.

Sebuah elemen masukan formulir yang nilainya diatur oleh React disebut sebagai *_controlled component*. Saat seorang pengguna memasukkan data kedalam _controlled component_, sebuah _event handler_ terpicu, dan kode Anda yang selanjutnya menentukan apakah masukan valid atau tidak (dengan cara merender ulang dengan nilai yang telah diperbarui). Jika Anda tidak merender ulang, maka nilai formulir tidak akan berubah.

> Sebuah *uncontrolled component* bekerja seperti hal yang dilakukan elemen formulir diluar React. Disaat pengguna memasukkan data kedalam formulir (melalui kotak input, _dropdown_, dll) nilai yang telah diperbarui akan digunakan tanpa React perlu melakukan apapun. Bagaimanapun, hal ini berari Anda tidak dapat memaksa sebuah bidang masukan unntuk memiliki nilai tertentu.

Reviewer: berari -> berarti typo 😄 

I think the translation of `However, this also means ...` would be closer to `Namun, hal ini berarti ...`
What do you think? 😄

Dalam banyak kasus, Anda harus menggunakan _controlled components_.

## [Keys](/docs/lists-and-keys.html) {#keys}

> Sebuah "key" adalah karakter spesial yang perlu disertakan saat membuat kumpulan elemen dalam bentuk array. Key membantu React untuk mengidentifikasi bagian mana yang telah diubah, ditambah, atau dihilangkan. Keys harus diberikan pada elemen dalam sebuah array untuk memberikan identitas yang stabil pada elemen tersebut.

Reviewer: ```suggestion
Sebuah "*key*" adalah atibut *string* spesial yang perlu disertakan saat membuat kumpulan elemen dalam bentuk senarai. *Key* membantu React untuk mengidentifikasi bagian mana yang telah diubah, ditambah, atau dihilangkan. *Keys* harus diberikan pada elemen dalam sebuah senarai untuk memberikan identitas yang stabil pada elemen tersebut.
```

> Keys hanya perlu bersifat unik dengan elemen lainnya dalam array yag sama. Keys tidak perlu unik diseluruh aplikasi atau bahkan didalam komponen yang sama.

Reviewer: yag -> yang typo 😄 
diseluruh -> di seluruh
didalam -> di dalam

> Jangan mengoper nilai `Math.random()` sebagai key. Penting untuk diingat bahwa key harus memiliki "identitas yang stabil" walau melalui dirender ulang, sehingga React dapat mentukan kapan sebuah item ditambah, dihapus, atau diurutkan. Secara ideal, key harus tersusun dari data Anda yang bersifat unik dan stabil, seperti `post.id`.

Reviewer: dirender -> di-*render*
mentukan -> menentukan

## [Refs](/docs/refs-and-the-dom.html) {#refs}

React mendukung atribut spesial yang dapat dilampirkan di komponen apapun. Atribut `ref` dapat terdiri dari sebuah objek yang dibuat oleh [`React.createRef()` function](/docs/react-api.html#reactcreateref) atau oleh sebuah _callback function_, atau _string_ (di API _legacy_). Disaat `ref` adalah sebuah _callback function_, fungsi tersebut dapat menerima argumen berupa elemen DOM atau sebuah kelas (bergantung pada tipe elemen). Hal ini dapat memberikan Anda akses langsung pada elemen DOM atau komponen yang berhubungan.

> Gunakan refs dengan bijak. Jika anda merasa sering menggunakan refs untuk "melakukan suatu hal", pertimbangkan untuk lebih familier dengan [aliran data top-down](/docs/lifting-state-up.html).

Reviewer: top-down -> *top-down*

## [Event](/docs/handling-events.html) {#events}

> Handling events with React elements has some syntactic differences:

Reviewer: Whoops, looks like you forgot to delete this 😄

> Menangani event dengan Elemen React memiliki perbedaan secara sintaksis:

Reviewer: event -> *event*

> * _Event handler_ pada React ditulis menggunakan camelCase, bukan sebagaii huruf kecil.

Reviewer: camelCase -> *camelCase*
sebagaii -> sebagai

* Dengan JSX, Anda mengoper sebuah fungsi sebagai _event handler_, bukan sebagai string.

## [Rekonsiliasi](/docs/reconciliation.html) {#reconciliation}

> Disaat sebuah _props_ atau _state_ dari sebuah komponen mengalami perubahan, React dapat memutuskan apakah hal tersebut memerlukan pembaruan DOM, dengan cara membandingkan elemen komponen yang dikembalikan dengan komponen yang telah dirender sebelumnya. Jika keduanya berbeda, React akan memperbaharui DOM. Proses ini disebut dengan istilah "rekonsiliasi".

Reviewer: dirender -> di-*render*
memperbaharui -> memperbarui 

memperbaharui is in the old EYD, while in PUEBi it had been changed to memperbarui 😄
